src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Templates/TemplateInserterStep.cs

```diff
@@ -71,7 +71,7 @@ public TemplateInserterStep(TemplateProvider templateProvider, ILogger<TemplateI
             }
         }
 
-        protected override bool IsApplicableImpl(IUpgradeContext context) => context?.CurrentProject is not null && _templateProvider.TemplateConfigFileNames.Any();
+        protected override bool IsApplicableImpl(IUpgradeContext context) => context?.CurrentProject is not null && context.CurrentProject.Language == Languages.CSharp && _templateProvider.TemplateConfigFileNames.Any();
```

Comment:
Also, like the TryConvertStep I think this check should happen in InitializeImpl so that the step always shows up in the list of needed upgrade steps and then is skipped (with a warning) if the project isn't SDK-style when we get to this step.